scipy/signal/wavelets.py

```diff
@@ -351,14 +453,20 @@ def cwt(data, wavelet, widths):
     >>> sig  = np.cos(2 * np.pi * 7 * t) + signal.gausspulse(t - 0.4, fc=2)
     >>> widths = np.arange(1, 31)
     >>> cwtmatr = signal.cwt(sig, signal.ricker, widths)
-    >>> plt.imshow(cwtmatr, extent=[-1, 1, 31, 1], cmap='PRGn', aspect='auto',
+    >>> plt.imshow(cwtmatr, extent=[-1, 1, 1, 31], cmap='PRGn', aspect='auto',
     ...            vmax=abs(cwtmatr).max(), vmin=-abs(cwtmatr).max())
     >>> plt.show()
-
     """
-    output = np.zeros([len(widths), len(data)])
+    # Determine output type
+    if dtype is None:
+        if np.asarray(wavelet(1, widths[0], **kwargs)).dtype.char in 'FD':
```

Comment:
maybe also `'G'` for complex256? Not sure if it's supported or not

Reply:
I don't see a reason why it wouldn't be supported, but I'm not sure how to test this on my machine. I've left it as it was for now.

Reply:
I would try adding it to the list, and also adding some test where you cast to `complex256` and see if you get the same result (to numerical precision) as the `complex128` result

Reply:
I've tested `cwt` with `complex256` a bit and found no apparent issues, so I've added 'G' to the list of wavelet dtypes.

A test like that would be tricky to implement I think, since `cwt` calls the wavelet function itself and checks the dtype of the output. There is no way of setting the dtype of the wavelet when calling `cwt` with any of the currently implemented wavelet functions, the `dtype` arguments only affects the output of `cwt`.
As far as I know the availability of `complex256` is also platform dependent - wouldn't this create problems on test machines where `complex256` is not available?

Reply:
> As far as I know the availability of complex256 is also platform dependent - wouldn't a test like that create problems on test machines where complex256 is not available?

Not sure, but we have code like this elsewhere so if we need to fix it at some point we can do it multiple places (including here)